### PR TITLE
Add filelogger

### PR DESCRIPTION
**Pull request by MarcelMue**
Still WIP a bit, especially naming is hard. It seems to work though.

towards https://github.com/giantswarm/giantswarm/issues/4029

### CODE REVIEW

File: pkg/framework/filelogger/filelogger.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	return f, nil
+}
+
+func (f FileLogger) StartLoggingPod(name, namespace string) error {
+	req := f.k8sClient.CoreV1().RESTClient().Get().Namespace(namespace).Name(name).Resource("pods").SubResource("log").Param("follow", strconv.FormatBool(true))
+	readCloser, err := req.Stream()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	go f.scan(readCloser, name)
```
Comment (MarcelMue) [inline pkg/framework/filelogger/filelogger.go]: Need to do this so everything else can keep running.

File: pkg/framework/filelogger/filelogger.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	defer outFile.Close()
+
+	f.logger.Log("level", "debug", "message", fmt.Sprintf("logging output of %s to %s", name, outFile.Name()))
+	_, err = io.Copy(outFile, readCloser)
+	if err != nil {
+		f.logger.Log("level", "error", "stack", microerror.Mask(err))
+	}
+
+	f.logger.Log("level", "debug", "message", fmt.Sprintf("logged output of %s to %s", name, outFile.Name()))
```
Comment (MarcelMue) [inline pkg/framework/filelogger/filelogger.go]: I can not see this log when I run the test, maybe I misunderstand something but I expected the `readCloser` to reach EOF before and then get here.

File: pkg/framework/host.go [hunk 3]
```diff
@@ -297,7 +311,21 @@ func (h *Host) InstallBranchOperator(name, cr, values string) error {
 }
 
 func (h *Host) InstallOperator(name, cr, values, version string) error {
-	return h.InstallResource(name, values, version, h.crd(cr))
+	err := h.InstallResource(name, values, version, h.crd(cr))
```
Comment (MarcelMue) [inline pkg/framework/host.go]: This is not nice, or pretty but it seems to work.

**Comment by MarcelMue** [comment]
I tested this on `chart-operator` - you can check one build with the log here:
https://circleci.com/gh/giantswarm/chart-operator/1429#artifacts/containers/0

The change to get the logs in circle is quite concise, check the PR here: https://github.com/giantswarm/chart-operator/pull/125

File: pkg/framework/filelogger/filelogger.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	}
+
+	return f, nil
+}
+
+func (f FileLogger) StartLoggingPod(name, namespace string) error {
+	req := f.k8sClient.CoreV1().RESTClient().Get().Namespace(namespace).Name(name).Resource("pods").SubResource("log").Param("follow", strconv.FormatBool(true))
+	readCloser, err := req.Stream()
+	if err != nil {
+		return microerror.Mask(err)
```
Comment (MarcelMue) [inline pkg/framework/filelogger/filelogger.go]: Need to retry here - the pod might still be in `container creating` :(

**Review by xh3b4sd — CHANGES_REQUESTED** [review]
Cool thing so far.

File: pkg/framework/filelogger/filelogger.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+	f := &FileLogger{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+	}
+
+	return f, nil
+}
+
+func (f FileLogger) StartLoggingPod(name, namespace string) error {
```
[the remark this answers is not in the document]
Comment (MarcelMue) [reply pkg/framework/filelogger/filelogger.go]: Done.

File: pkg/framework/filelogger/filelogger.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+	outFile, err := os.Create(logfilepath)
+	if err != nil {
+		f.logger.Log("level", "error", "stack", microerror.Mask(err))
+		return
+	}
+
+	defer outFile.Close()
+
+	f.logger.Log("level", "debug", "message", fmt.Sprintf("logging output of %s to %s", name, outFile.Name()))
```
[the remark this answers is not in the document]
Comment (MarcelMue) [reply pkg/framework/filelogger/filelogger.go]: Done.

File: pkg/framework/host.go [hunk 7]
```diff
@@ -297,7 +312,21 @@ func (h *Host) InstallBranchOperator(name, cr, values string) error {
...
+		return err
+	}
+	podName, err := h.PodName("giantswarm", fmt.Sprintf("app=%s", name))
+	if err != nil {
+		return err
+	}
+
+	err = h.filelogger.StartLoggingPod(podName, "giantswarm")
+	if err != nil {
+		return err
```
[the remark this answers is not in the document]
Comment (MarcelMue) [reply pkg/framework/host.go]: Done.

**Review by xh3b4sd — APPROVED** [review]

**Pull request merged**